### ISSUE TITLE
Custom inputs not working for me.

### ISSUE THREAD

**Issue by anonymous-tech-user** [mmende/homebridge-samsungtv-control2#5]
I installed and set up the new alpha plug in and it works well with my 2011 Samsung TV but I had trouble adding the custom inputs. No matter what I put, it only shows me the current source inputs and this Youttube input which does nothing on mine since YouTube was removed with a S/W update. I don't mind having the current source inputs (They are very useful) but I cannot get rid of the Youttube nor can I add other inputs (Channels, CC, etc.)
Below is my config.json if there are any errors:
{
            "devices": [
                {
                    "usn": "xxxxxxx",
                    "name": "Home TV",
                    "lastKnownIp": "192.168.1.xxx",
                    "mac": "xx:xx:xx:xx:xx",
                    "delay": 400,
                    "inputs": [
                        {
                            "name": "TV",
                            "keys": "KEY_TV"
                        },
                        {
                            "name": "MENU",
                            "keys": "KEY_MENU"
                        }
                    ],
                    "disableUpnpSetters": false
                }
            ],
            "platform": "SamsungTVControl"
        }

**Comment by mmende** [comment]
Did you restart homebridge after the edits? Furthermore it should be required to remove the TV and then add it again since homekit sees stuff like input sources as static I guess. Also at the current version when restarting homebridge remember to keep your tv turned on because the plugin doesn't remember previously seen devices over restarts yet.

**Comment by anonymous-tech-user** [comment]
I removed the TV from the home app, and restarted Homebridge with edits while the TV was on and now I can't add the TV to HomeKit anymore, it simply doesn't find it. Although under the Homebridge logs, the TV has been initialized and configured and I can see it properly under the accessories page. It just won't be added to HomeKit now.

**Comment by mmende** [comment]
You can try to remove the accessory file for the tv before restarting homebridge. That solved this kind of problem for me at least. In the homebridge ui click on the three dots in the upper right corner and then `Terminal`. Then get a list of accessory files in the persist folder `ls persist/`. This should list a bunch of files like `AccessoryInfo.1B1DA7AF38C0.json`. You can then have a look in to these files with `more persist/AccessoryInfo.1B1DA7AF38C0.json`. One of these files should contain the name of your tv `{"displayName":"Home TV"...`. If you found this file just rename it like this `mv persist/AccessoryInfo.1B1DA7AF38C0.json persist/AccessoryInfo.1B1DA7AF38C0.json.backup` and restart homebridge.

**Comment by anonymous-tech-user** [comment]
Thank you, the custom inputs finally work, that trick to remove accessory file helped get it back in order. Only two things now, could you please show me how to remove that, “youttube” (YouTube) input? It doesn’t do anything and it is a little irritating. Secondly, you mentioned how the tv has to be on when homebridge restarts but even after homebridge has been set up and device info is mentioned (uuid, mac, ip) it does not update once the TV has been switched on after being turned off unless homebridge is restarted. Is there a way around that? Perhaps adding a ping response (or something like that, not sure what it’s called) to see the TV status is on or off every 30 seconds might help get around this? 
Once again, thanks a lot, this plug in is really good

**Comment by mmende** [comment]
The rc library allows opening some apps (on some models) like youtube in a different way then sending keys to the tv. Currently some of them (e.g. youtube) will be added hardcoded for testing purposes mainly. I will publish a new release where you can configure which of these apps should appear in the inputs. The on / off state management is also in my todo list. For now it should be possible after turning the tv on manually to click the power toggle in the home app... afterwards the tv updated it's state in my home app and I could control it again

**Comment by anonymous-tech-user** [comment]
Oh I see, well thank you again. And as for the refreshing by turning on the TV manually and then clicking in the home app, that doesn't seem to work, the Home app only reports as the TV "updating" for a while before it just says no response- Clicking it doesn't do anything other than giving a haptic. Weird enough, when I first configured it, the Home app would refresh the status of the TV but after I removed it/added it and removed the accessory file, this did not go back to normal and now unless I restart Homebridge while the TV is on, it doesn't seem to work at all.
